demo/Menu.tsx

```diff
@@ -0,0 +1,70 @@
+import { toggleMark } from "prosemirror-commands";
+import type { MarkType } from "prosemirror-model";
+import type { EditorState } from "prosemirror-state";
+import React, { ReactNode } from "react";
+
+import { useEditorEventCallback, useEditorState } from "../src/index.js";
+
+function isMarkActive(mark: MarkType, state: EditorState): boolean {
+  if (!state.selection) return false;
+  const { from, $from, to, empty } = state.selection;
+  return empty
```

Comment:
I wonder if you need the ternary or if this would work, regardless of selection emptiness: `mark.isInSet(state.storedMarks) || state.doc.rangeHasMark(from, to, mark)`.

It's possible that changing the selection clears the stored marks anyway.

Reply:
the code is lifted from the prosemirror-example-setup (added a comment):
https://github.com/ProseMirror/prosemirror-example-setup/blob/master/src/menu.ts#L58

it has the same condition, but i don't know why 🤷

Reply:
Cool.